### PR TITLE
Keeper Security - New Elastic Integration Package

### PR DESCRIPTION
**Pull request by jpkeepersecurity**
## Proposed commit message

**Add Keeper Security agentless integration package**

This PR introduces a new Keeper Security integration that enables **truly agentless** audit event collection by allowing Keeper to push data directly to Elasticsearch via the Bulk API without requiring any Elastic Agent installation.

**WHAT**: 
- New integration package `keeper_security_siem_integration` with complete ECS field mapping
- Agentless architecture: Keeper Security Platform → Elasticsearch Bulk API → Ingest Pipeline → Index  
- Comprehensive ingest pipeline with timestamp normalization, GeoIP enrichment, and ECS compliance
- Pre-built Kibana dashboard for audit event visualization
- Support for multiple timestamp formats and robust error handling
- Built-in field mappings for user information, source IPs, organization data, and related entities

**WHY**: 
- Eliminates the operational overhead of installing and managing Elastic Agents
- Provides real-time audit event processing directly from Keeper Security platform
- Ensures consistent ECS field mapping for better integration with Elastic Security workflows
- Offers immediate value with pre-configured dashboards and visualizations
- Supports enterprise security monitoring with minimal infrastructure requirements

The integration processes Keeper audit events (authentication, security actions, user management) and enriches them with geographic location data, user correlation fields, and standardized ECS categorization.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [x] Integration package structure follows Elastic standards with proper manifest, changelog, and documentation
- [x] Ingest pipeline includes comprehensive error handling and ECS field mapping
- [x] Data stream configuration supports dynamic datasets and namespaces
- [x] Documentation includes complete setup instructions with API key creation and GeoIP configuration
- [x] Dashboard provides meaningful visualizations for audit event analysis
- [x] Field mappings include related.* fields for proper entity correlation

## How to test this PR locally

1. **Install the integration assets in Kibana:**
   ```bash
   # Navigate to Management > Integrations in Kibana
   # Search for "Keeper Security" and install assets only
   ```

2. **Create API key for data ingestion:**
   ```bash
   POST /_security/api_key
   {
     "name": "keeper-integration-test",
     "expiration": "365d", 
     "role_descriptors": {
       "keeper-writer": {
         "cluster": ["monitor"],
         "indices": [
           {
             "names": ["logs-keeper.audit-*"],
             "privileges": ["auto_configure", "create_doc"]
           }
         ]
       }
     }
   }
   ```

3. **Enable GeoIP enrichment:**
   ```bash
   PUT /_cluster/settings
   {
     "persistent": {
       "ingest.geoip.downloader.enabled": true
     }
   }
   ```

4. **Test data ingestion:**
   ```bash
   curl --location 'https://YOUR_ELASTICSEARCH_HOST/logs-keeper.audit-1.0.0/_bulk' \
   --header 'Authorization: ApiKey YOUR_API_KEY' \
   --header 'Content-Type: application/x-ndjson' \
   --data-raw '{"create":{}}
   {"audit_event":"test_login","remote_address":"8.8.8.8","username":"test@company.com","enterprise_id":1234,"timestamp":"2025-08-28T17:36:11.621Z","client_version":"Web.1.0.0","category":"authentication"}
   '
   ```

5. **Verify in Kibana:**
   - Check data in Discover with index pattern `logs-keeper.audit-*`
   - View the "Keeper SIEM Integration - Dashboard" 
   - Verify ECS field mapping and GeoIP enrichment

## Related issues

- Addresses enterprise customer requirements for agentless security event collection
- Supports Keeper Security platform integration with Elastic SIEM workflows

## Screenshots

image
image

The integration includes a pre-built dashboard showing:
- Audit event timeline and volume
- Geographic distribution of access attempts  
- User activity patterns and authentication events
- Event categorization and outcome analysis

### CODE REVIEW

**Comment by efd6** [comment]
@jpkeepersecurity Please sign the CLA with all the email addresses used as author email in the commits in this PR.

**Comment by jpkeepersecurity** [comment]
I just signed @efd6

File: packages/keeper_security_siem_integration/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- version: "1.0.0"
+  changes:
+    - description: Initial release of Keeper Security agentless integration
+      type: enhancement
+      link: ""
```
Comment (efd6) [inline packages/keeper_security_siem_integration/changelog.yml]:
```suggestion
      link: https://github.com/elastic/integrations/pull/15089
```

File: packages/keeper_security_siem_integration/changelog.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+- version: "1.0.0"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/changelog.yml]: Do you want this GA in the initial release?

File: packages/keeper_security_siem_integration/_dev/build/build.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+dependencies:
+  ecs:
+    reference: git@v8.11.0
```
Comment (efd6) [inline packages/keeper_security_siem_integration/_dev/build/build.yml]:
```suggestion
    reference: git@v8.17.0
```

File: packages/keeper_security_siem_integration/docs/README.md
Comment (efd6) [inline packages/keeper_security_siem_integration/docs/README.md:1]: This should be done in _dev/build/docs/README.md and generated to here.

Please also take a look at the guidelines [here](https://www.elastic.co/docs/extend/integrations/documentation-guidelines) and make the documentation match that structure.

File: packages/keeper_security_siem_integration/img/keeper-dashboard.png
Comment (efd6) [inline packages/keeper_security_siem_integration/img/keeper-dashboard.png:1]: This image looks odd; the aspect ratio is off. Can you fix it please?

File: packages/keeper_security_siem_integration/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    type: image/png
+icons:
+  - src: /img/keeper-logo.svg
+    title: Keeper Security
+    size: 32x32
+    type: image/svg+xml
+# Required owner field with type
+owner:
+  github: elastic/security-external-integrations
+  type: elastic
```
Comment (efd6) [inline packages/keeper_security_siem_integration/manifest.yml]: This would be either "partner" or "community".

File: packages/keeper_security_siem_integration/data_stream/audit/fields/base-fields.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      value: "logs"
+      description: Data stream type
+    - name: dataset
+      type: constant_keyword
+      value: "keeper.audit"
+      description: Data stream dataset
+    - name: namespace
+      type: constant_keyword
+      value: "default"
+      description: Data stream namespace
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/fields/base-fields.yml]: Please use `external: ecs` here instead of re-specifying.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+      field: timestamp
+      ignore_missing: true
+
+on_failure:
+  - append:
+      field: error.message
+      value: "Failed to process Keeper audit event: {{{_ingest.on_failure_message}}}"
+  - set:
+      field: event.kind
+      value: pipeline_error
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:137]:
```suggestion
  - set:
      field: event.kind
      tag: set_pipeline_error_into_event_kind
      value: pipeline_error
      if: ctx.error?.message != null
  - append:
      field: tags
      value: preserve_original_event
      allow_duplicates: false
      if: ctx.error?.message != null
on_failure:
  - append:
      field: error.message
      value: >-
        Processor '{{{ _ingest.on_failure_processor_type }}}'
        {{{#_ingest.on_failure_processor_tag}}}with tag '{{{ _ingest.on_failure_processor_tag }}}'
        {{{/_ingest.on_failure_processor_tag}}}failed with message '{{{ _ingest.on_failure_message }}}'
  - set:
      field: event.kind
      tag: set_pipeline_error_to_event_kind
      value: pipeline_error
  - append:
      field: tags
      value: preserve_original_event
      allow_duplicates: false
```

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+      if: ctx.keeper?.timestamp != null
+      on_failure:
+        - append:
+            field: error.message
+            value: "Failed to parse timestamp: {{{keeper.timestamp}}}"
+
+  # If no keeper.timestamp, try to use existing @timestamp or set current time
+  - set:
+      field: '@timestamp'
+      value: "{{_ingest.timestamp}}"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]:
```suggestion
      value: "{{{_ingest.timestamp}}}"
```
use triple stache throughout

Though this processor does not need to be included since the `@timestamp` will be set by fleet if it is not present, and including this here means that the product of the pipeline tests will not be resproducible.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 8]
```diff
@@ -0,0 +1,137 @@
+---
+# data_stream/audit/elasticsearch/ingest_pipeline/default.yml (FIXED)
+description: "Agentless pipeline for processing Keeper Security audit events"
+processors:
+  # Set data stream information
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: We should set the `ecs.version` early.

**Review by efd6 — COMMENTED** [review]
Please add pipeline and system tests.

**Review by efd6 — COMMENTED** [review]
Please make the following change:
```diff
diff --git a/.github/CODEOWNERS b/.github/CODEOWNERS
index e54e89dd94..0e842c8e83 100644
--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -290,6 +290,7 @@
 /packages/juniper_srx @elastic/integration-experience
 /packages/kafka @elastic/obs-infraobs-integrations
 /packages/kafka_log @elastic/obs-infraobs-integrations
+/packages/keeper_security_siem_integration @elastic/security-service-integrations
 /packages/keycloak @elastic/security-service-integrations
 /packages/kibana @elastic/stack-monitoring
 /packages/kubernetes @elastic/obs-ds-hosted-services
```

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml (hunk 7 above)
Comment (andrewkroh) [reply packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: With `set` you can use `copy_from` instead of the Mustache templates.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+      value: "{{_ingest.timestamp}}"
+      if: ctx['@timestamp'] == null
+
+  # Set user information from username
+  - set:
+      field: user.name
+      value: "{{username}}"
+      if: ctx.username != null
+  
+  # If username looks like email, set user.email
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: These comments can go into the `description` field. Then they become part of the pipeline specification and are visible in other places that you can access the pipeline definition, like the Ingest Pipeline editor/viewer.

File: packages/keeper_security_siem_integration/data_stream/audit/fields/base-fields.yml [hunk 10]
```diff
@@ -0,0 +1,51 @@
+---
+# data_stream/audit/fields/base-fields.yml (REQUIRED @timestamp field)
+- name: '@timestamp'
+  type: date
+  description: Date/time when the audit event occurred
+- name: message
+  type: wildcard
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/data_stream/audit/fields/base-fields.yml]: Fields that are defined in ECS should use the ECS definition.

https://github.com/elastic/integrations/wiki/Fleet-Package-Code-Review-Comments#defining-an-ecs-field-without-using-an-external-definition

File: packages/keeper_security_siem_integration/manifest.yml [hunk 11]
```diff
@@ -0,0 +1,31 @@
+format_version: 3.0.0
+name: keeper
+title: "Keeper Security"
+version: 1.0.0
+description: >
+  Keeper Security agentless integration for collecting audit events directly  via Elasticsearch Bulk API. No agents required - Keeper pushes data directly  to Elasticsearch ingest pipelines.
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/manifest.yml]: There are double spaces.

```suggestion
  Keeper Security agentless integration for collecting audit events directly via Elasticsearch Bulk API. No agents required - Keeper pushes data directly to Elasticsearch.
```

Does this description render fully or is it truncated? It's longer than most packages' descriptions.

File: packages/keeper_security_siem_integration/data_stream/audit/manifest.yml [hunk 12]
```diff
@@ -0,0 +1,6 @@
+title: "Keeper Audit Events"
+type: logs
+dataset: keeper.audit
+elasticsearch:
+  dynamic_dataset: true
+  dynamic_namespace: true
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/data_stream/audit/manifest.yml]: Why are these necessary? The pipeline sets static values for both dataset and namespace. And these are primarily used to give the Elastic Agent permissions to write to a wide swath of `logs-*-*`, but this integration advertises itself as not needing an Elastic Agent.

File: packages/keeper_security_siem_integration/_dev/build/build.yml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+dependencies:
+  ecs:
+    reference: git@v8.17.0
+    import_mappings: true
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/_dev/build/build.yml]: This shouldn't be needed any more. All integrations receive the ecs@mappings template which provides similar functionality as `import_mappings: true`.

File: packages/keeper_security_siem_integration/_dev/build/build.yml [hunk 14]
```diff
@@ -0,0 +1,4 @@
+dependencies:
+  ecs:
+    reference: git@v8.17.0
```
Comment (andrewkroh) [inline packages/keeper_security_siem_integration/_dev/build/build.yml:3]: There are newer ECS versions. Can we use the latest version?

File: .github/CODEOWNERS [hunk 15]
```diff
@@ -293,6 +293,7 @@
 /packages/juniper_srx @elastic/integration-experience
 /packages/kafka @elastic/obs-infraobs-integrations
 /packages/kafka_log @elastic/obs-infraobs-integrations
+packages/keeper_security_siem_integration @elastic/security-service-integrations
```
Comment (efd6) [inline .github/CODEOWNERS]:
```suggestion
/packages/keeper_security_siem_integration @elastic/security-service-integrations
```

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+  source.geo.country_iso_code: ".*"
+  source.geo.country_name: ".*"
+  source.geo.location.lat: "^-?\\d+\\.\\d+$"
+  source.geo.location.lon: "^-?\\d+\\.\\d+$"
+  source.geo.region_iso_code: ".*"
+  source.geo.region_name: ".*"
+  source.geo.city_name: ".*"
+  # Timestamps might have slight variations in parsing
+  "@timestamp": "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{3})?Z$"
+
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml]: I don't see that any of these are necessary.

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+  source.geo.country_iso_code: ".*"
+  source.geo.country_name: ".*"
+  source.geo.location.lat: "^-?\\d+\\.\\d+$"
+  source.geo.location.lon: "^-?\\d+\\.\\d+$"
+  source.geo.region_iso_code: ".*"
+  source.geo.region_name: ".*"
+  source.geo.city_name: ".*"
+  # Timestamps might have slight variations in parsing
+  "@timestamp": "^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{3})?Z$"
+
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml]: I don't think any of this is necessary.

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+numeric_keyword_fields:
+  # Enterprise ID should be treated as string in Elasticsearch
+  - organization.id
+  - enterprise_id
+
+# Note: ignore_missing_exception is not a valid configuration option
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml]: What is this for?

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+numeric_keyword_fields:
+  # Enterprise ID should be treated as string in Elasticsearch
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml]: Can you clarify this so that we can get rid of the comment?

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-timestamp-formats.json-config.yml [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+  # GeoIP data can vary based on database updates
+  source.geo.continent_name: ".*"
+  source.geo.country_iso_code: ".*"
+  source.geo.country_name: ".*"
+  source.geo.location.lat: "^-?\\d+\\.\\d+$"
+  source.geo.location.lon: "^-?\\d+\\.\\d+$"
+  source.geo.region_iso_code: ".*"
+  source.geo.region_name: ".*"
+  source.geo.city_name: ".*"
+
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-timestamp-formats.json-config.yml]: These should not be necessary.

File: packages/keeper_security_siem_integration/manifest.yml [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    size: 600x400
+    type: image/png
+icons:
+  - src: /img/keeper-logo.svg
+    title: Keeper Security
+    size: 32x32
+    type: image/svg+xml
+# Required owner field with type
+owner:
+  github: elastic/security-external-integrations
```
Comment (efd6) [inline packages/keeper_security_siem_integration/manifest.yml]:
```suggestion
  github: elastic/security-service-integrations
```

**Review by efd6 — COMMENTED** [review]
There are outstanding queries that need to be addressed:
- [ ] https://github.com/elastic/integrations/pull/15089/files#r2323636518
- [ ] https://github.com/elastic/integrations/pull/15089/files#r2323666366
- [ ] https://github.com/elastic/integrations/pull/15089/files#r2323665477
- [ ] https://github.com/elastic/integrations/pull/15089/files#r2335240729

**Comment by jpkeepersecurity** [comment]
@efd6 
 https://github.com/elastic/integrations/pull/15089/files#r2323636518 ✅ 
 https://github.com/elastic/integrations/pull/15089/files#r2323666366 ✅
 https://github.com/elastic/integrations/pull/15089/files#r2323665477 ✅
 https://github.com/elastic/integrations/pull/15089/files#r2335240729 ✅

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 22]
```diff
@@ -2,6 +2,10 @@
 # data_stream/audit/elasticsearch/ingest_pipeline/default.yml (FIXED)
 description: "Agentless pipeline for processing Keeper Security audit events"
 processors:
+  # Set ECS version
+  - set:
+      field: ecs.version
+      value: "8.11"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: 8.17.0

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-edge-cases.json-config.yml:1]: This should not be necessary.

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-timestamp-formats.json-config.yml
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-timestamp-formats.json-config.yml:1]: This should not be necessary.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 23]
```diff
@@ -0,0 +1,163 @@
...
+        - append:
+            field: error.message
+            value: "Failed to parse timestamp: {{{keeper.timestamp}}}"
+ 
+  - set:
+      field: "@timestamp"
+      copy_from: "_ingest.timestamp"
+      if: ctx['@timestamp'] == null
+      description: "If no keeper.timestamp, try to use existing @timestamp or set current time"
+
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]:
```suggestion
```
This is handled by fleet and just makes testing more difficult if included.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+  - set:
+      field: user.name
+      value: "{{username}}"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: All the instances of `{{<somefield>}}` should be `{{{<somefield>}}}`.

The basis for this is that the double-stache interprets the value as HTML that needs to be escaped. We don't want that.

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json-config.yml:1]: This is not necessary.

**Comment by jpkeepersecurity** [comment]
pr is up to date.

File: packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+      "remote_address": "172.16.0.25",
+      "audit_event": "set_two_factor_off",
+      "category": "security",
+      "client_version": "EMConsole.18.0.0",
+      "username": "user4@company.com",
+      "enterprise_id": 1088,
+      "timestamp": "2025-09-26T06:46:05.810Z"
+    }
+  ]
+}
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/_dev/test/pipeline/test-sample.json]: Add final new line.

File: packages/keeper_security_siem_integration/docs/README.md
Comment (efd6) [inline packages/keeper_security_siem_integration/docs/README.md:1]: This should be done in _dev/build/docs/README.md and generated to here.

Please also take a look at the guidelines [here](https://www.elastic.co/docs/extend/integrations/documentation-guidelines) and make the documentation match that structure.

[ref](https://github.com/elastic/integrations/pull/15089#discussion_r2323640280)

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 26]
```diff
@@ -0,0 +1,157 @@
...
+      value: ["authentication", "web"]
+  - set:
+      field: event.type
+      value: ["access", "info"]
+  - set:
+      field: event.dataset
+      value: "keeper.audit"
+  - set:
+      field: event.module
+      value: "keeper"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml]: These can be `constant_keyword`s and be set statically in the fields definitions.

File: packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 27]
```diff
@@ -0,0 +1,157 @@
...
+      field: user.email  
+      value: "{{{username}}}"
+      if: ctx.username != null && ctx.username.contains("@")
+      description: "If username looks like email, set user.email"
+
+  - set:
+      field: source.ip
+      value: "{{{remote_address}}}"
+      if: ctx.remote_address != null
+      description: "Set source IP from remote_address"
```
Comment (efd6) [inline packages/keeper_security_siem_integration/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:56]:
```suggestion
  - set:
      field: user.name
      copy_from: username
      ignore_empty_value: true
      description: "Set user name from username"

  - set:
      field: user.email  
      copy_from: username
      if: ctx.username != null && ctx.username.contains("@")
      description: "If username looks like email, set user.email"

  - set:
      field: source.ip
      copy_from: remote_address
      ignore_empty_value: true
      description: "Set source IP from remote_address"
```
similar below

**Comment by jpkeepersecurity** [comment]
Currently, for agentless integrations of the push type, implementing system tests is not required, since during deployment elastic-package looks for elastic-agent environment variables by default.

However, given that there is now a new agentless push integration, I suggest updating elastic-package to support system tests for this type of solution. Once the tool provides this support, the integration can then include system tests as part of its setup.

**Review by efd6 — APPROVED** [review]
